net/url/url.h

```diff
@@ -247,9 +247,12 @@ struct URLParametersExtractor {
         return default_value;
       }
     }
+    const std::map<std::string, std::string>& AsImmutableMap() const { return parameters_; }
     std::map<std::string, std::string> parameters_;
   };
 
+  const std::map<std::string, std::string>& AllQueryParameters() const { return query.AsImmutableMap(); }
```

Comment:
`const_cast` is not ever require here.

Why not just `{ return query.parameters_; }` is a good question. Guess I wanted to play a good citizen and not access potentially private parts :)